composites/AlgoliaSearch/AlgoliaSearcher.js

```diff
@@ -141,182 +168,147 @@ class AlgoliaSearcher extends React.Component {
+		}
 
 		return (
-			<div className="wpseo-kb-search-detail">
-				<div className="wpseo-kb-search-navigation">
-					<button className="button dashicon-button wpseo-kb-search-back-button"
-					        aria-label={ this.props.backLabel }
-					        onClick={ this.hideDetail }>
-						{ this.props.back }
-					</button>
-
-					<a href={ post.permalink }
-					   className="button dashicon-button wpseo-kb-search-ext-link "
-					   aria-label={ this.props.openLabel }
-					   target="_blank">
-						{ this.props.open }
-					</a>
-				</div>
-
-				<ArticleContent post={post} iframeTitle={this.props.iframeTitle}/>
-			</div>
+			<SearchBar
+				submitAction={ this.onSearchButtonClick.bind( this ) }
+				searchString={ this.state.searchString }
+			/>
 		);
 	}
 
 	/**
-	 * Log any occuring error and render a search error warning.
+	 * Gets the current error message if it is set in the state.
 	 *
-	 * @param {string} errorMessage The message to display.
-	 * @returns {ReactElement} A p tag with a warning that the search was not completed.
+	 * @returns {ReactElement|string} Returns a rendered error object if an error is set.
 	 */
-	renderError( errorMessage ) {
-		console.error( errorMessage );
-		a11ySpeak( this.props.errorMessage );
+	getErrorMessage() {
+		if ( this.state.errorMessage ) {
+			return this.renderError();
+		}
 
-		return ( <p>{ this.props.errorMessage }</p> );
+		return null;
 	}
 
 	/**
-	 * Creates the Search Bar component with additional components such as a loading indicator, errors etc.
+	 * Gets the loading indicator.
 	 *
-	 * @returns {ReactElement} A div containing the search bar and potential other components.
+	 * @returns {ReactElement|string} Returns a loader if the searching state is active.
 	 */
-	createSearchBar() {
-		let errorMessage = "";
-		let loadingIndicator = "";
-		let resultsHeading = "";
-		let results = "";
-
-		let searchBar = <SearchBar
-			headingText={ this.props.headingText }
-			submitAction={ this.searchButtonClicked.bind( this ) }
-			searchString={ this.state.searchString }
-			searchButtonText={ this.props.searchButtonText }
-		/>;
-
-		// Show an error message.
-		if ( this.state.errorMessage ) {
-			errorMessage = this.renderError( this.state.errorMessage );
+	getLoadingIndicator() {
+		// Don't show loading indicator when a search has already been performed, to avoid flickering.
+		if ( this.state.searching && ! this.state.results ) {
+			const loadingPlaceholder = this.props.intl.formatMessage( messages.loadingPlaceholder );
+			return <Loading placeholder={ loadingPlaceholder } />;
 		}
 
-		// Show a loading indicator.
-		if ( this.state.searching ) {
-			loadingIndicator = <Loading loadingPlaceholder={ this.props.loadingPlaceholder } />;
+		return null;
+	}
+
+	/**
+	 * Determines what the search results view needs to look like.
+	 *
+	 * @returns {ReactElement|string} Returns a specific search result object based on state.
+	 */
+	determineSearchResultsView() {
```

Comment:
in this function there are 3 if and a default return, not sure it there are other cases to take into account apart from `this.state.currentDetailViewIndex === -1` and `this.state.currentDetailViewIndex !== -1`. Mayne this could be a bit optimized?